app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -0,0 +1,131 @@
+package org.oppia.app.player.exploration
+
+import android.app.Application
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.idling.CountingIdlingResource
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asCoroutineDispatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import java.util.concurrent.AbstractExecutorService
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/** Tests for [ExplorationActivity]. */
+@RunWith(AndroidJUnit4::class)
+class ExplorationActivityTest {
+  @Before
+  fun setUp() {
+    IdlingRegistry.getInstance().register(MainThreadExecutor.countingResource)
```

Comment:
Out of curiosity, is this IdlingResource actually needed for your test? If not, I suggest removing it & the MainThreadExecutor. Prefer not adding extra test infrastructure if not needed.

Reply:
Even if I run `ExplorationAcitvityTest`,  `HomeActivityTest` will run and this error occurs.

Reply:
Ah, this relates to #97. Similarly to how we set up blocking dispatcher, we need to set add these new Logger booleans to the test module for HomeActivityTest so that the're properly set.

Or is this also broken on develop? There's a chance I broke it with #104, but I haven't noticed that yet. It may be the case that this code is now introducing a Logger dependency in HomeActivity, and since HomeActivityTest uses a different root application component we need to make sure all dependencies are satisfied. This will get much easier once #59 is done.

Reply:
I did check that in `develop` right now and it is failing there also.

Reply:
Ack, thanks! I'll follow up with a fix shortly.

Reply:
#131 is now out for review.

Reply:
Removed unnecessary code of IdlingResource and MainThreadExecutor